lib/pure/strinterp.nim

```diff
@@ -0,0 +1,320 @@
+#
+#
+#            Nim's Runtime Library
+#        (c) Copyright 2012 Andreas Rumpf
+#
+#    See the file "copying.txt", included in this
+#    distribution, for details about the copyright.
+#
+
+## This module provides the string interpolation macro ``fmt``.
+
+import parseutils, sequtils, macros, strutils
+
+# -----------------------------------------------------------------------------
+# nimboost's formatters
+# -----------------------------------------------------------------------------
+
+proc mkDigit(v: int, lowerCase: bool): string {.inline.} =
+  doAssert(v < 26)
+  if v < 10:
+    result = $chr(ord('0') + v)
+  else:
+    result = $chr(ord(if lowerCase: 'a' else: 'A') + v - 10)
+
+proc intToStr(n: SomeNumber, radix = 10, len = 0, fill = ' ', lowerCase = false): string =
```

Comment:
Can't this function be implemented in terms of the existing functions for converting an int to a string? It seems to me that this reinvents a lot of the code that we already have causing duplication.

Reply:
On a closer look, there is not as much duplication as it seems. The part that can be rewritten is line 43 to 49, but it would be slightly less efficient, because it would create several temporary strings and requires two passes for the case conversion (`toHex` only allows to produce upper case). Note that there is probably not much room for simplification for the sign placement logic at the bottom because it depends on whether "zero filling" is desired. I have a few smaller things though which makes the code slightly more concise, which I'll push.